sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/AppServiceBaseImpl.java

```diff
@@ -205,15 +204,11 @@ public PublishingProfile getPublishingProfile() {
     }
 
     public Mono<PublishingProfile> getPublishingProfileAsync() {
-        return FluxUtil
-            .collectBytesInByteBufferStream(
-                manager()
-                    .serviceClient()
-                    .getWebApps()
-                    .listPublishingProfileXmlWithSecretsAsync(
-                        resourceGroupName(), name(), new CsmPublishingProfileOptions()))
-            .map(
-                bytes -> new PublishingProfileImpl(new String(bytes, StandardCharsets.UTF_8), AppServiceBaseImpl.this));
+        return manager()
+            .serviceClient()
+            .getWebApps()
+            .listPublishingProfileXmlWithSecretsAsync(resourceGroupName(), name(), new CsmPublishingProfileOptions())
+            .map(binaryData -> new PublishingProfileImpl(binaryData.toString(), this));
```

Comment:
`Flux<ByteBuffer>` is replaced by `BinaryData` in mgmt as well?

Reply:
Yes.

If for GA SDK there be breaking changes, I believe there is a flag to revert to older behavior.